### PR TITLE
[Documentation] Docs Contribution Guide update

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | yes
| New docs?    | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Awesome, thank you Magda! 👍

File: docs/contributing/documentation/overview.rst [hunk 1]
```diff
@@ -33,14 +41,43 @@ Create a dedicated branch for your changes (for organization):
...
-you're done, push this branch to *your* GitHub fork and initiate a pull request.
+You can now make your changes directly to this branch and commit them.
+Remember to name you commits descriptively, keep them possibly small, with just unitary changes (such that change something only in one part of the docs, not everywhere).
+
+When you're done, push this branch to *your* GitHub fork and initiate a pull request.
+
+Your pull request will be reviewed, you will be asked to apply fixes if necessary and then it will be merged into the main repository.
+
+
+Testing Documentation
```
Comment (pamil) [inline docs/contributing/documentation/overview.rst]: IMO this section should reflect https://github.com/Sylius/Sylius/tree/master/docs#build.

Comment (TheMadeleine) [inline docs/contributing/documentation/overview.rst]: Or the other way: The readme should reflect the documentation :)

File: docs/contributing/documentation/format.rst [hunk 2]
```diff
@@ -135,66 +135,28 @@ Using the path and filename of the page without the extension, for example:
...
-.. _documents:               https://github.com/Sylius/Sylius-Docs
-.. _reStructuredText Primer: http://sphinx-doc.org/rest.html
-.. _markup:                  http://sphinx-doc.org/markup/
-.. _Pygments website:        http://pygments.org/languages/
-.. _source:                  https://github.com/fabpot/sphinx-php
-.. _Sphinx quick setup:      http://sphinx-doc.org/tutorial.html#setting-up-the-documentation-sources
+.. _`documents`:               https://github.com/Sylius/Sylius/tree/master/docs
+.. _`reStructuredText Primer`: http://www.sphinx-doc.org/en/stable/rest.html
+.. _`markup`:                  http://www.sphinx-doc.org/en/stable/markup/
+.. _`Pygments website`:                  http://pygments.org/languages/
```
Comment (michalmarcinkowski) [inline docs/contributing/documentation/format.rst]: Wrong indent.

**Pull request merged**